File: 66. Plus One.cpp
#include <iostream>
#include <vector>
using namespace std;
class Solution
{
public:
    vector<int> plusOne(vector<int> &digits)
    {
        int tens = 0;
        cout << *digits.end() << endl;
        // for (int i = digits.size() - 1; i >= 0; i--)
        // {
        //     int plus_one = digits[i] + 1 + tens;
        //     int tens = plus_one / 10;
        //     int sigle = plus % 10;

        //     if (tens > 0)
        //     {
        //     }
        // }
        return digits;
    }
};

int main()
{
    cout << "hello world!" << endl;
    vector<int> test = {1, 0, 9};
    Solution *s = new Solution();
    s->plusOne(test);
    return 0;
}

File: 66. Plus One.ts
export function plusOne(digits: number[]): number[] {
    if (digits.length < 1) return [];

    let length = digits.length,
        plusOne: number = digits[length - 1] + 1,
        tens: number = Math.floor(plusOne / 10),
        sigle: number = plusOne % 10;
    digits[length - 1] = sigle;
    for (let i = length - 2; i >= 0; i--) {
        console.log(i);

        let sum = digits[i] + tens;
        tens = Math.floor(sum / 10);
        sigle = sum % 10;
        digits[i] = sigle;
        if (tens < 1) break;
    }
    if (tens > 0) digits.unshift(tens);
    return digits;
    // return String(Number(digits.join("")) + 1)
    //     .split("")
    //     .map((val) => Number(val));
}

let t = [
    [1, 2, 3],
    [4, 3, 2, 1],
    [9],
    [6, 1, 4, 5, 3, 9, 0, 1, 9, 5, 1, 8, 6, 7, 0, 5, 5, 4, 3],
];

for (const c of t) {
    let res = plusOne(c);
    console.log(res);
    console.log("-------------------");
}
